File: order/src/main/java/com/trinhvu/order/viewmodel/order/OrderGetVm.java
package com.trinhvu.order.viewmodel.order;

import com.trinhvu.order.model.Order;

import java.math.BigDecimal;
import java.time.ZonedDateTime;
import java.util.List;

public record OrderGetVm(
        Long id,
        BigDecimal total,
        List<OrderItemGetVm> orderItems,
        ZonedDateTime createdOn
) {
    public static OrderGetVm fromModel(Order order) {
        return new OrderGetVm(
                order.getId(),
                order.getTotalPrice(),
                OrderItemGetVm.fromModels(order.getOrderItems()),
                order.getCreatedOn()
        );
    }
}

File: stock-watch-fe/components/order/models/OrderGetVm.ts
import { OrderItemGetVm } from "./OrderItemGetVm";

export interface OrderGetVm {
  id: number;
  total: number; // BigDecimal represented as number in TypeScript
  orderItems: OrderItemGetVm[];
  createdOn: string; // ZonedDateTime as ISO string (e.g., "2023-10-15T14:48:00Z")
}
